### DISCUSSION TITLE
How to use subscription data in other thread?

### DISCUSSION THREAD

**Discussion by JungminHaMartin** [FreeOpcUa/python-opcua#1432] [Q&A]
I subscribe some nodes and I want to use the value in other thread.
How I deal with the subscripted data?
Please show the code as below.

class SubHandler(object):

    def datachange_notification(self, node, val, data):

        dict[node] = val
        Thread_Status.node_value = val

    def event_notification(self, event):
        print("Python: New event", event)

class Thread_Status(QThread):
    def __init__(self, parent):
        super().__init__(parent)
        self.parent = parent
        self.node = []
        self.node_value = []

**Comment by AndreasHeine** [comment]
in python threads have shared state -> so a dict containing the variables and values with a threadlock should do the trick!

**Comment by oroulet** [comment]
or put them in threading.Queue() object., depending what fits you best, They lock for you.
